crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -126,70 +126,88 @@ impl RuffSettingsIndex {
         }
 
         // Add any settings within the workspace itself
-        let mut walker = WalkDir::new(root).into_iter();
-
-        while let Some(entry) = walker.next() {
-            let Ok(entry) = entry else {
-                continue;
-            };
-
-            // Skip non-directories.
-            if !entry.file_type().is_dir() {
-                continue;
-            }
-
-            let directory = entry.into_path();
+        let mut builder = WalkBuilder::new(root);
+        builder.standard_filters(true);
```

Comment:
Do we need to respect the `respect_gitignore` setting here?

Reply:
I've added it where by default it's `true` and if we find any config file it'll use the value from that instead.